fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -376,14 +376,14 @@ public CommandProcessingResult deposit(final Long savingsId, final JsonCommand c
 
             LOG.debug("Deposit account has been created: {} ", deposit);
 
-            GroupSavingsIndividualMonitoring gsim = gsimRepository.findById(account.getGsim().getId()).get();
+            GroupSavingsIndividualMonitoring gsim = gsimRepository.getById(account.getGsim().getId());
             LOG.info("parent deposit : {} ", gsim.getParentDeposit());
             LOG.info("child account : {} ", savingsId);
             BigDecimal currentBalance = gsim.getParentDeposit();
             BigDecimal newBalance = currentBalance.add(transactionAmount);
             gsim.setParentDeposit(newBalance);
             gsimRepository.save(gsim);
-            LOG.info("balance after making deposit : {} ", gsimRepository.findById(account.getGsim().getId()).get().getParentDeposit());
+            LOG.info("balance after making deposit : {} ", gsimRepository.getById(account.getGsim().getId()).getParentDeposit());
```

Comment:
This is anyway a bad pattern because we're making a database query just to log something.
What if the logger is not even configured to info level but above, like warn? Then the logger won't log anything but we still make the db call.

In those cases it's wise to wrap the log message into a level check like:

```
if (LOG.isInfoEnabled()) {
    // do the expensive logging
}
```